### ISSUE TITLE
Pluralized View Guessing, and Controllers?

### ISSUE THREAD

**Issue by Zaggen** [balderdashy/sails#2909]
I ask this as a community question. A similar question was asked before [issues-645](https://github.com/balderdashy/sails/issues/645), but this is somehow different. I want to know why sails is following a singular name convention for controllers and views instead of pluralized names like in [rails](https://github.com/railstutorial/sample_app/tree/master/app). I think it makes sense to have a model in singular to represent an entity, but usually a controller can handle a collection of models and pluck just one if necessary, so in theory the controller will be a controlling multiple models, hence the name should be pluralized, that's usually what   you want in a route too, at least as default. Now, you can make a generator to to pluralize your controller name, but my concern here is the default behavior of `res.ok` will try to guess the view path if none provided when a request  does not specify for a json response, so sails will check for a singular path.

```javascript
// lib/hooks/views/res.view.js
// Try to guess the view by looking at the controller/action
if (!req.options.view && (req.options.controller || req.options.model)) {
    relPathToView = (req.options.controller||req.options.model) + '/' + (req.options.action || 'index');
}
```

This means, if i'm not mistaken, that if you want to keep the functionality of blueprints to give a view based on your controller/model name, you will have name your view folders in singular, so instead of having:
```batch
/views
├── services
│  ├── index.ejs
│  ├── ...
```
We have to use
```batch
/views
├── service
│  ├── index.ejs
│  ├── ...
```

I'm just wondering if there is an underlying reason to adopt this singular named constrollers/view convention instead of the pluralized one?
If models are agnostic, then shouldn't the view guessing behavior also try to check for pluralized folders too? In case somebody wants to use the pluralized convention?
I hope my question makes sense, and that i did't say any stupidity (but i probably did), let me know what you guys think.

**Comment by sailsbot** [comment]
Thanks for posting, @Zaggen.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
